components/atoms/TextInput/text-input.tsx

```diff
@@ -14,6 +14,10 @@ interface TextInputProps {
   classNames?: string;
   errorMsg?: string;
   value?: string;
+  defaultValue?: string;
+  required?: boolean;
+  innerRef?: any;
```

Comment:
Is there no generic that can be used here? `fieldRef` would be a better name also.

Reply:
A generic is a type that you can specify, similar to `React.useRef<HTMLFormElement>` where the `T` in `React.useRef<T>()` is the generic you provide to give additional context to the shape of what you're providing.